File: haskell/fibonacci.hs
-- Find the nth Fibonacci number starting at 0

-- base cases 0 and 1 and otherwise Naive solution
    fib :: Int -> Int 
    fib 0 = 0
    fib 1 = 1 
    fib n = fib (n - 1) + fib (n - 2)

File: csharp/Fibonacci.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FibonacciCSharp
{
    class Program
    {
        static int fibonacci(int n)
        {
            int[] f = new int[n + 1];
            int i;

            f[0] = 0;
            f[1] = 1;

            for(i = 2; i <= n; i++)
            {
                f[i] = f[i - 1] + f[i - 2];
            }

            return f[n];
        }

        static void Main(string[] args)
        {
            Console.WriteLine("Write the nth term for which you'd like to find the fibonacci number for: ");
            int number = Convert.ToInt32(Console.ReadLine());

            int sum = fibonacci(number);

            Console.WriteLine(sum);
            Console.ReadKey();
        }
    }
}
